### PR TITLE
[DEN-88] Formatting changes, code cleanup, WAF Scopes Create

### PR DESCRIPTION
**Pull request by stevenpaz**

### CODE REVIEW

File: ec/helper/tf.go [hunk 1]
```diff
@@ -98,7 +75,100 @@ func NewTerraformSet(items []interface{}) *schema.Set {
...
+			return nil
+		}
+		return &i
+	}
+	return nil
+}
+
+// ExpandBoolPointer converts an interface{} to *bool
+// If v is nil or not a bool, a nil pointer is returned
+func ExpandBoolPointer(v interface{}) *bool {
```
Comment (changgyu-oh) [inline ec/helper/tf.go:102]: Expand- doesn't sound right. ConvertTo or CastTo sounds better.

File: examples/resources/ec_waf_scopes/resource.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    name           = "Scopes Test"
+    
+    host {
+      is_case_insensitive = false
+      type                = "EM"
+      values              = ["site1.com/path1","site2.com"]
+    }
+
+    limit {
+      id                   = "<Rate Rule ID #1>"
```
Comment (changgyu-oh) [inline examples/resources/ec_waf_scopes/resource.tf:14]: Is it generated by server?
If so, no need to be here?

Comment (stevenpaz) [reply examples/resources/ec_waf_scopes/resource.tf:14]: "Limit" refers to Rate rules, so this is a Rate Rule ID. So it is already a separate resource. The naming is just not consistent on the WAF API side.

File: ec/helper/tf.go (hunk 1 above)
Comment (stevenpaz) [reply ec/helper/tf.go]: Updated

**Review by wdawg33 — APPROVED** [review]
Great work, @stevenpaz

**Pull request merged**